packages/edit-site/src/hooks/commands/use-navigation-commands.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * WordPress dependencies
+ */
+import { privateApis } from '@wordpress/commands';
+import { __ } from '@wordpress/i18n';
+import { useMemo } from '@wordpress/element';
+import { useSelect } from '@wordpress/data';
+import { store as coreStore } from '@wordpress/core-data';
+
+/**
+ * Internal dependencies
+ */
+import { store as editSiteStore } from '../../store';
+import { unlock } from '../../private-apis';
+import { useHistory } from '../../components/routes';
+
+const { useCommandLoader } = unlock( privateApis );
+
+const getNavigationCommandLoaderPerPostType = ( postType ) =>
+	function useNavigationCommandLoader( { search } ) {
+		const supportsSearch = ! [ 'wp_template', 'wp_template_part' ].includes(
+			postType
+		);
+		const deps = supportsSearch ? [ search ] : [];
+		const history = useHistory();
+		const { canvasMode, records, isLoading } = useSelect( ( select ) => {
+			const { getEntityRecords } = select( coreStore );
+			const query = supportsSearch
+				? {
+						search,
+						per_page: 10,
+						orderby: 'date',
+				  }
+				: {
+						per_page: -1,
+				  };
+			return {
+				records: getEntityRecords( 'postType', postType, query ),
+				isLoading: ! select( coreStore ).hasFinishedResolution(
+					'getEntityRecords',
+					[ 'postType', postType, query ]
+				),
+				canvasMode: unlock( select( editSiteStore ) ).getCanvasMode(),
+			};
+		}, deps );
+
+		const commands = useMemo( () => {
+			return ( records ?? [] ).slice( 0, 10 ).map( ( record ) => {
```

Comment:
Is the slicing necessary, given that the query uses `per_page`?

Reply:
the templates page doesn't use per page

Reply:
@youknowriad I'm also curious about this `.slice`. I understand why it's here given the templates endpoint doesn't have a `per_page` to limit return results, but as far as I am aware it doesn't have an `order_by` either [so we're slicing relevant items from the records](https://github.com/WordPress/gutenberg/issues/51657).

If the templates records are the only ones we're slicing, and we have them from the response anyway could we potentially:

1. Order them based on relevance here, before slicing them.
2. Remove the slice altogether.

Thanks!